### ISSUE TITLE
Child lock

### ISSUE THREAD

**Issue by edwardtfn** [Blackymas/NSPanel_HA_Blueprint#1727]
1. Option to set a separate color for the thermostat icon and the status when the heating is active. Thermostat / Heat - ICON COLOR (Optional) split separately.

2. **Be able to lock the screen. Child lock or something similar. Entering some code or holding down some combination of virtual buttons.**

Thanks.

_Originally posted by @elemarek in https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/572_

**Comment by acca84** [comment]
Hello, 
I'm interested in feature 2 
Any chance to add it in the near future?
Thanks

**Comment by MiKlassen** [comment]
Childlock would be very great.
Maybe hold both buttons for 3 secs?

**Comment by edwardtfn** [comment]
What shoukd child lock do?
1. Block the screen to wake-up?
2. Block the buttons to do anything?
3. Should child lock activates onky when screensaver is active or shoukd it be independently (most likely)?

**Comment by MiKlassen** [comment]
Childlock needs to be independent of screensaver but maybe a option to enable it along with screensaver? 

Regarding to the feature set 

For best ux I would think of something like this

If childlock active 
  Wake up on all Input (Button and Touch) to 
  show Message Screen with "ChildlockMsg" + "DeactivationHint"
  Sleep

**Comment by elemarek** [comment]
My idea is that: When the child lock is activated, only the hardware buttons function, and the display lights up upon touch, allowing for screen navigation. However, it should not be possible to make any changes via screen touch, such as switching software buttons on/off or similar actions. Essentially, it would be in a so-called read-only mode. On the home screen, a key icon would be lit, and touching it would require a code to unlock it into write mode.

**Comment by edwardtfn** [comment]
So, if I understood correctly, we should have a separated timer for the child lock, and the rest work just as it is (fallback to home, then to screensaver, when enabled, etc.).
Then the child lock timer goes to 0, the panel enters on the child lock mode and all the other timers keeps its previous behavior.
When in child lock, regardless on which screen it is, if the screen is touched it should display a keyboard where users could type a pre-defined code to unlock (code should come from a sensor, from blueprint or hard-coded on ESPHome?). When unlocked, the panel should follow the behavior as if no child lock, until that timer goes to 0 again.
If no code is typed on the keyboard after some time, it should fall back to the previous state.
Hardware buttons functionality during child lock should be optional.

Would a first implementation totally based on `substitutions` and customizations be good enough? Like an add-on?

**Comment by MiKlassen** [comment]
As "Addon" would be good enough. 

But the timer behavior is not really suitable I think at least not with my kids 😂. 
I would need a way to actively enable it and also when they are sleeping I don't really want to enter the code everytime I use the panel. 

The code would be a nice to have,  but maybe over the top. 
Thinking of other devices in kitchen etc. with such a feature. Those only use a simple combination of two or three buttons to enable/disable. 

Therefor was my simple idea. 
Lock and Unlock it via 3sec Hold of both buttons. 
If locked just show "Lockscreen" and a Hint to unlock. 

Personally I think a "readonly mode" like mentioned above could lead to irritating behaviour.

**Comment by edwardtfn** [comment]
But what one would be able to do and not do while the panel is "child locked"?

**Comment by MiKlassen** [comment]
While locked no interaction possible, only unlock. 

Home and Screensaver normal visble but Input blocked.
If touche , show Message Screen with "Locked"

**Comment by edwardtfn** [comment]
So the only difference for what I suggested is not going to lock automatically …

**Comment by MiKlassen** [comment]
Yes most likely and Code should be optional.

If no code is set unlock should be the same way as lock

**Comment by elemarek** [comment]
For me, the essence of a child lock is to prevent the child from being able to turn on or change something I don't want, meaning they shouldn't be able to control the panel. If the child lock mode merely displayed the time and a prompt to unlock, the entire panel would lose its purpose. A child should be able to view information, but I don't want them to be able to control it.

I have panels in every room, including the children's room. From time to time, I would like the option to turn something on when I'm in their room, but I don't want the child to be able to control it, while simultaneously allowing them to read all the data from the panel. I see no reason for the display to become just a dumb time indicator when the child lock is active. Currently, I had to remove everything I don't want them to control, but it would be useful for me to keep those controls there.

Activating and deactivating the lock should be handled with a numeric code, and the locking or unlocking could be controlled within the menu where the panel's brightness is set.

**Comment by MiKlassen** [comment]
I see your point and can relate to that. I would call that Kiosk mode it would be far more than a childlock it would be suitable for public room etc. 

If your child's are old enough to read data from it I see the point for that usage. 

For a small one that is a capable of pushing a chair to it and just "gamble" around on the panel a locked mode would more be suitable - at this point it would for me and my family more irritating to swipe but can't control anything.

**Comment by elemarek** [comment]
It would be sufficient for me if, in child lock mode, it displayed at least the main screen without overlaying any message, so that all data such as temperature, weather, and so on are still readable. Alternatively, provide the option to enter the Entities States menu.

There could be a small lock icon; when I long-press it, a window pops up to enter the code. The same action could be used to lock the panel again. This would immediately make it clear that the panel is locked.

The lock could also have an option to select which functions are unlocked and which are locked. For example, the child could still control the heating in their room, but they would be unable to control functions like turning off all lights in the entire house.

**Comment by edwardtfn** [comment]
But the panel isn’t aware if a specific entity is on the same room or if that is the entire house. Home Assistant can’t provide area, so we know what is in the same area as the panel, but it is impossible to know if an entity is related to the entire house.

**Comment by elemarek** [comment]
Yes, I understand, I just gave an example. Which entity would be accessible while the child lock is active could be defined in the Blueprint using a checkbox for each entity I define. I am referring to entities for buttons and other controls and similar. To start with, simply blocking interaction with the display would be enough, but at the same time, maintaining the possibility to read and display data. The unlock code could also be defined via the Blueprint.

**Comment by DunklerPhoenix** [comment]
It would be also great to hide some actions behind a pin code.
I want to use NSPanel in our gym so some options like the temperature control should be locked behind a pin code or something :D

**Comment by DunklerPhoenix** [comment]
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed in 14 days if no further activity occurs. Thank you for your contributions!

ewwwwwww
